presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixClient.java

```diff
@@ -0,0 +1,283 @@
+        try {
+            return connection.unwrap(PhoenixConnection.class);
+        }
+        catch (Exception e) {
+            try (Connection closingConnection = connection) {
+                // use try with resources to close properly
+            }
+            throw new PrestoException(PHOENIX_QUERY_ERROR, "Couldn't open Phoenix connection", e);
+        }
+    }
+
+    public org.apache.hadoop.hbase.client.Connection getHConnection()
+            throws IOException
+    {
+        return HBaseFactoryProvider.getHConnectionFactory().createConnection(hbaseConfig);
+    }
+
+    public void execute(ConnectorSession session, String statement)
+    {
+        try (Connection connection = connectionFactory.openConnection(JdbcIdentity.from(session))) {
+            execute(connection, statement);
+        }
+        catch (SQLException e) {
+            throw new PrestoException(PHOENIX_QUERY_ERROR, "Error while executing statement", e);
+        }
+    }
+
+    @Override
+    public Set<String> getSchemaNames(JdbcIdentity identity)
+    {
+        ImmutableSet.Builder<String> schemaNames = ImmutableSet.builder();
+        schemaNames.add(DEFAULT_SCHEMA);
+        schemaNames.addAll(super.getSchemaNames(identity));
+        return schemaNames.build();
+    }
+
+    @Override
+    protected SchemaTableName getSchemaTableName(ResultSet resultSet)
+            throws SQLException
+    {
+        return new SchemaTableName(
+                toPrestoSchemaName(resultSet.getString(TABLE_SCHEM)),
+                resultSet.getString(TABLE_NAME));
+    }
+
+    @Override
+    public PreparedStatement buildSql(ConnectorSession session, Connection connection, JdbcSplit split, List<JdbcColumnHandle> columnHandles)
+            throws SQLException
+    {
+        PreparedStatement query = super.buildSql(session, connection, split, columnHandles);
+        QueryPlan queryPlan = getQueryPlan((PhoenixPreparedStatement) query);
+        try {
+            // PhoenixRecordReader constructs the ResultSet internally
+            RecordReader<NullWritable, PhoenixRecordWritable> reader = new PhoenixRecordReader<>(PhoenixRecordWritable.class, new JobConf(), queryPlan);
+            reader.initialize(((PhoenixSplit) split).getPhoenixInputSplit(), null);
+            return new DelegatePreparedStatement(query)
+            {
+                @Override
+                public ResultSet executeQuery()
+                        throws SQLException
+                {
+                    try {
+                        return (ResultSet) resultSetField.get(reader);
+                    }
+                    catch (IllegalAccessException e) {
+                        throw new PrestoException(PHOENIX_INTERNAL_ERROR, e);
+                    }
+                }
+            };
+        }
+        catch (IOException | InterruptedException e) {
+            throw new PrestoException(PHOENIX_QUERY_ERROR, "Error while setting up Phoenix ResultSet", e);
```

Comment:
Interrupted exception is not an error. Other than that why `InterruptedException` is thrown here.